appium/webdriver/appium_service.py

```diff
@@ -215,38 +119,42 @@ def start(self, **kwargs: Any) -> sp.Popen:
             error_msg = startup_failure_msg
         if error_msg is not None:
             if stderr == sp.PIPE and self._process.stderr is not None:
+                # noinspection PyUnresolvedReferences
                 err_output = self._process.stderr.read()
                 if err_output:
                     error_msg += f'\nOriginal error: {str(err_output)}'
             self.stop()
             raise AppiumServiceError(error_msg)
         return self._process
 
-    def stop(self) -> bool:
+    def stop(self, timeout: float = 5.5) -> bool:
         """Stops Appium service if it is running.
 
         The call will be ignored if the service is not running
         or has been already stopped.
 
-        Returns:
-            `True` if the service was running before being stopped
+        :param timeout: The maximum time in float seconds to wait
+        for the server process to terminate
+        :return: `True` if the service was running before being stopped
         """
-        is_terminated = False
+        was_running = False
         if self.is_running:
             assert self._process
+            was_running = True
             self._process.terminate()
-            self._process.communicate(timeout=5)
-            is_terminated = True
+            try:
+                self._process.communicate(timeout=timeout)
+            except sp.SubprocessError:
+                self._process.kill()
```

Comment:
actually, after kill, we need wait/communicate
```suggestion
                self._process.kill()
                _, _ = self._process. communicate(timeout=timeout)
```
(or print the error) to ensure the process state

Reply:
kill sends SIGKILL, which terminates the process unconditionally.
maybe there is some official link to docs about the best way of cross-platform graceful/forceful process termination ?

Reply:
I haven't found out cross platform thing, but my comment to put communicate comes from https://docs.python.org/3/library/subprocess.html#subprocess.Popen.communicate 's example code. Actually it is not after terminate's exception though.

On Windows, I explicitly called taskkill below with force option when terminate/kill fails.

```
os.system("taskkill /f /pid {}".format(process.pid))
process.wait(timeout=5)
```

Based on my past Win's experience in work, the taskkill helped to kill process forcefully. The terminate/kill in Python was a graceful one. Probably... this taskkill could be ok to add when we get a new issue though

Reply:
it's actually a good idea to use taskkill, I'll add it to the implementation